python/looker_sdk/rtl/api_settings.py

```diff
@@ -46,25 +47,26 @@ def _convert_bool(val: str, _: bool) -> bool:
     return converted
 
 
-@attr.s(auto_attribs=True, kw_only=True)
-class ApiSettings(transport.TransportSettings):
-    """API Configuration Settings.
+class PApiSettings(transport.PTransportSettings, Protocol):
+    filename: str
```

Comment:
I feel like `filename` is an implementation detail `ApiSettings`. It used to be `_filename`, was there a reason to promote it?

and now that I think about it, same thing for `section` (sorry for the back adn forth on that). `filename` and `section` are particular to reading a config file and don't belong in the more general `PApiSettings` type that other (db? redis? dynamodb?) config settings might implement.

Reply:
`_filename` was promoted because we are assigning it outside of the class in `configure`:

```python
@classmethod
def configure(
    cls, filename: str = "looker.ini", section: Optional[str] = None
) -> PApiSettings:
    api_settings = cls(filename=filename, section=section)
    .
    .
    .
    settings = converter.structure(config_data, ApiSettings)
```